deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -928,6 +928,7 @@ def get_train_op(self):
         optimizer = self.graph.optimizer
       else:
         optimizer = self.optimizer
+      # Should we keep a separate global step count for each submodel?
```

Comment:
For progressive model, using a shared global step doesn't seem natural(in case of learning rate decay), defining a separate parameter might be better.